src/audio_to_image_model/audio_to_iamge_standalone_predict.py

```diff
@@ -39,21 +39,36 @@ def predict(model, audio_segments, device):
     return np.array(predictions)
 
 
+# def generate_csv(predictions, threshold, row_id_prefix, label_to_bird_map):
+#    result_rows = []
+#    for i, pred in enumerate(predictions):
+#        #print(f"Prediction {i}: {pred}")
+#        max_index = np.argmax(pred[0])
+#        # birds_present = {label_to_bird_map[index]: 1 if p >= threshold else 0 for index, p in enumerate(pred[0])}
+#        birds_present = {label_to_bird_map[index]: 1 if index == max_index else 0 for index, p in enumerate(pred[0])}
+#        row_id = f"{row_id_prefix}_{i * 5 + 5}"
+#        result_rows.append([row_id] + list(birds_present.values()))
+#    # Change the column names to bird names
+#    column_names = ['row_id'] + [label_to_bird_map[i] for i in range(0, len(label_to_bird_map))]
+#    result_df = pd.DataFrame(result_rows, columns=column_names)
+#    return result_df
+
+
 def generate_csv(predictions, threshold, row_id_prefix, label_to_bird_map):
     result_rows = []
-
     for i, pred in enumerate(predictions):
-        print(f"Prediction {i}: {pred}")
-        birds_present = [1 if p >= threshold else 0 for p in pred[0]]
-        row_id = f"{row_id_prefix}_{i * 5}"
-        result_rows.append([row_id] + birds_present)
-
-    result_df = pd.DataFrame(result_rows, columns=['row_id'] + [f'bird{i}' for i in range(1, len(label_to_bird_map) + 1)])
+        # print(f"Prediction {i}: {pred}")
+        max_index = np.argmax(pred[0])
+        birds_present = {label_to_bird_map[index]: pred[0][index] for index, p in enumerate(pred[0])}
+        row_id = f"{row_id_prefix}_{i * 5 + 5}"
+        result_rows.append([row_id] + list(birds_present.values()))
+    # Change the column names to bird names
+    column_names = ['row_id'] + [label_to_bird_map[i] for i in range(0, len(label_to_bird_map))]
+    result_df = pd.DataFrame(result_rows, columns=column_names)
     return result_df
 
 
-
-def inference(audio_path, model, label_to_bird_map, threshold=0.6, device='cpu'):
+def inference(audio_path, model, label_to_bird_map, threshold=0.5017, device='cpu'):
```

Comment:
Why is it 0.5017? Where does the number come from?

Reply:
The value is not being used. I will delete it.